vignettes/runif-via-pmap.Rmd

```diff
@@ -0,0 +1,213 @@
+---
+title: "Generate data from different distributions via pmap()"
+author: Safouane Chergui
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(purrr)
+```
+
+### Generate numbers with different parameters  via `runif()`
+
+Use `pmap()` in order to iterate over several arguments rowwise. This function comes in handy when one needs to generate numbers from a distribution while passing different argument each time the function gets called.
+
+The following chunk of code r `runif(n, min = 0, max = 1)` generates n numbers from the uniform distribution. The same operation can be done for several triples. 
+The first step is to store triples in a dataframe and then iterate over rows.
+
+```{r include=FALSE}
+library(tidyverse)
+```
+
+```{r}
+# Notice how df's variable names are the same as runif's argument names.
+df <- tribble(
+  ~ n, ~ min, ~ max,
+   1L,     0,     1,
+   2L,    10,   100,
+   3L,   100,  1000
+)
+df
+```
+
+Let us first set the seed and practice on single rows. Setting the seed will make the result reproducible and practicing on single rows will allow us to check if the results we get with `pmap()` are what we truly expect to get.
+
+```{r}
+set.seed(123)
+(x <- df[1, ])
+
+runif(n = x$n, min = x$min, max = x$max)
+
+x <- df[2, ]
+runif(n = x$n, min = x$min, max = x$max)
+
+x <- df[3, ]
+runif(n = x$n, min = x$min, max = x$max)
+
+```
+
+Now, let us try to do the same thing with `pmap()`.
+
+```{r}
+set.seed(123)
+pmap(df, runif)
+```
+
+### Finessing variable and argument names
+
+It is not mandatory for variable names to be the same as the argument names of the function. This can be arranged on the fly.In the example below, we do exactly this.
```

Comment:
This example feels a bit forced. Is there a more natural scenario in which renaming the variables is needed? Maybe you could simulate from a different distribution?